client/src/store/utils/thunkCreators.js

```diff
@@ -111,6 +135,22 @@ export const postMessage = (body) => async (dispatch) => {
   }
 };
 
+const updateReadStatus = async (unreadMessages) => {
+     await axios.put(`/api/messages/`, {read: true , unreadMessages});
+};
+
+export const updateReadMessages = (conversation, userId) => async (dispatch) => {
+  const unreadMessages = conversation.messages.filter(message=>{
+    if (!message.read && message.senderId === conversation.otherUser.id)
+        return message.id
+  }).map(message=>message.id);
```

Comment:
Instead of looping to find all unread messages ids you could simply pass the conversation ID and handle the logic of knowing which messages to update in the backend with the help of the `update()` function.

Reply:
Same answer as above